kmip/pie/api.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2015 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class KmipClientAPI:
+    """
+    A simplified KMIP client interface for conducting KMIP operations.
+
+    The KmipClientAPI provides a simple external interface for various KMIP
+    operations and composes the bulk of the PyKMIP Pie API.
+    """
+    @abc.abstractmethod
+    def create(self, algorithm, length, *args, **kwargs):
```

Comment:
Let's remove the args and kwargs. The problem is that if a caller is using a generic KmipClientAPI and passes those arguments then only certain instances can utilize them. This can cause inconsistent behavior if the KmipClient is swapped out with something else.